SatisfactoryModLoader/assets/BPInterface.cpp

```diff
@@ -0,0 +1,541 @@
+#include "stdafx.h"
+
+#include <assets/BPInterface.h>
+#include <util/Objects/UProperty.h>
+#include <util/Objects/FName.h>
+#include <util/Objects/FFrame.h>
+#include <util/Objects/UFunction.h>
+#include <util/Objects/UClass.h>
+
+#include <Windows.h>
+#include <detours.h>
+
+#include <../SatisfactorySDK/SDK.hpp>
+
+using namespace SML::Objects;
+
+namespace SML {
+	namespace Paks {
+		// detour function cache //
+		void(*constructUFunction)(UFunction*&, FFunctionParams) = nullptr;
+		void(*registerFunction)(UClass*, const char*, void*) = nullptr;
+		void(*addFunctionToMap)(UClass*, const FClassFunctionLinkInfo*, std::uint32_t) = nullptr;
+		void(*objCompInDefer)(Objects::UClass*(*)(), Objects::UClass*(*)(), const TCHAR*, const TCHAR*, bool, const TCHAR*, void(*)()) = nullptr;
+		void(*processNewObjs)() = nullptr;
+		void(*constructUClass)(Objects::UClass*&, FClassParams) = nullptr;
+		void(*privateStaticClassBody)(const TCHAR*, const TCHAR*, Objects::UClass*&, void(*)(), std::uint32_t, Objects::EClassFlags, Objects::EClassCastFlags, const TCHAR*, void(*)(FObjectInitializer&), Objects::UObject*(*)(FVtableHelper&), void(*)(Objects::UObject*, FReferenceCollector&), Objects::UClass*(*)(), Objects::UClass*(*)(), bool);
+
+		int getVtableSize(void** vtable) {
+			if (vtable)
+				for (int i = 0;; i++)
+					if (!vtable[i])
+						return i;
+
+			return 0;
+		}
```

Comment:
Please follow the code guidelines by using brackets.

Reply:
function made by @Brabb3l but I can change it ^^